ADS-485/blaze-retheming

j6ll opened the pull request:
Part of #ADS-485 and ADS-526

## Proposed Changes

These are visual updates to the Blaze dashboard to emulate a more WooCommerce-like feel. The main changes include bringing everything into cards and also to introduce a top bar. This is not an exhaustive set of changes but more a first step as we plan to add Blaze into the marketing channels.

## Why are these changes being made?

We plan to add more entry points from WooCommerce to Blaze and add Blaze as a marketing channel. This will bring the plugin in line with the other marketing channel interfaces. 

Starting point for reference:

<img width="1123" height="1265" alt="Screenshot 2025-11-19 at 16 21 52" src="https://github.com/user-attachments/assets/81c341fd-db4c-4f56-8f04-7dc3467834fe" />

## Testing Instructions

Test the core pages of the dashboard:
- Posts `/advertising/posts/site.com`
- Campaigns `/advertising/campaigns/site.com`
- Campaign `/advertising/campaigns/123/site.com`
- Payments `/advertising/payments/site.com`
- Receipts `/advertising/payments/receipt/123/site.com`


<img width="1118" height="1240" alt="Screenshot 2025-11-19 at 16 13 15" src="https://github.com/user-attachments/assets/6f84d9f4-a642-4ffa-8f7b-bb483ee6820a" />


Test with Jetpack
- Run `yarn dev --sync` from `apps/blaze-dashboard` this will copy the Jetpack dash to your sandbox
- Point  `widgets.wp.com` to your sandbox in `/etc/hosts`
- Clear caches and you should see the dashboard change
<img width="1127" height="793" alt="Screenshot 2025-11-20 at 11 40 19" src="https://github.com/user-attachments/assets/b4fff82d-2a2c-4ba6-82e5-638305554b14" />


Here is an example of "before" and "after". You'll see the top bar and cards I mentioned above and also the layout is generally a little bit more compact than previously. 

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

j6ll committed:
Re-theme into cards
Theming top bar and mobile styles

therocket-gr commented:
i noticed a minor fix, the border in the top area is rendered 2 times

<img width="115" height="226" alt="Screenshot 2025-11-21 at 15 31 28" src="https://github.com/user-attachments/assets/e6660edb-d2e3-4835-80ae-07ddac209804" />

i think we should add an extra rule in `.campaigns-total-stats__item` for the last child to not have border-right

therocket-gr commented:
minor comment. but i see the payment button in trunk is to the right

<img width="1139" height="170" alt="Screenshot 2025-11-21 at 15 43 52" src="https://github.com/user-attachments/assets/ba1bb840-e7b0-449d-a342-9cfc375dd13d" />
in this brnach i see it left aligned
<img width="639" height="186" alt="Screenshot 2025-11-21 at 15 43 56" src="https://github.com/user-attachments/assets/c2b46d55-5399-4190-b5a0-78c6698b4265" />


finally i think we should better align this text.  (before it was in a white bg so it didnt matter. but now that it is in a boxed frame it seems not properly aligned (vertically) 

<img width="795" height="114" alt="Screenshot 2025-11-21 at 15 44 17" src="https://github.com/user-attachments/assets/df66e0ec-b4d7-4720-9b6b-6557390e0911" />

we could also remove the padding from the bottom here too

so that this
<img width="1042" height="126" alt="Screenshot 2025-11-21 at 15 46 55" src="https://github.com/user-attachments/assets/9eea3853-95bc-44a4-93fa-ff43c349ab9b" />

becomes like this

<img width="1072" height="135" alt="Screenshot 2025-11-21 at 15 46 59" src="https://github.com/user-attachments/assets/7e135ce4-f540-4d6e-9091-09528862c561" />

sbarbosa reviewed (approved):
The code looks good @j6ll.

I found some minimal design issues, but I'm not sure if they were included in this PR or already in prod, so feel free to leave them for later on a different ticket.

The border on the tabs for the Woo version of the dashboard appears cut off:
<img width="1218" height="498" alt="Screenshot 2025-11-24 at 1 55 17 PM" src="https://github.com/user-attachments/assets/fe09811b-9abf-4a99-9c79-a00dd0ef1968" />

The menu for the tabs, in mobile also appears cut off:
<img width="514" height="590" alt="Screenshot 2025-11-24 at 1 58 03 PM" src="https://github.com/user-attachments/assets/d9beb633-0193-40dd-900c-f60d76b20652" />

When you select an option on the filter dropdown, the label appears to be taking more space and moves the other label, hiding the dropdown icon:
<img width="432" height="389" alt="Screenshot 2025-11-24 at 1 58 45 PM" src="https://github.com/user-attachments/assets/46245af2-0539-47c1-b0e3-b151b56b5bb0" />

As mentioned, these minor designs issues may already be in prod, so feel free to ignore them. We can open a new ticket to fix them in a follow up PR.

j6ll committed:
- Testing fixes
- Reinstate credit balance
- Fixing some overcomplicated layout styling and removing importance

j6ll commented:
Hey @sbarbosa & @therocket-gr, thank you for your initial reviews - I've addressed these and quite a few other new things.

Noticeable addition is credits balance - the rest are mostly layout bugs or alignment inconsistencies
<img width="1121" height="258" alt="Screenshot 2025-12-03 at 17 47 08" src="https://github.com/user-attachments/assets/0a15c378-ec97-483d-b54b-544a3bac8790" />

<img width="464" height="129" alt="Screenshot 2025-12-03 at 18 06 53" src="https://github.com/user-attachments/assets/6d913e3b-4cbe-4a32-b9e6-0f8a39865fd1" />

You'll see there's 2 commits in this branch - everything since the last review is in the 2nd commit 👍

sbarbosa reviewed (approved):
LGTM 👍🏻

j6ll committed:
Some alignment and minor fixes to spacing and table borders

j6ll committed:
Payments list inherited some styles that broke the layout

therocket-gr reviewed (approved):
tested locally, everything looks good and all sections seem to work perfectly well
code changes LGTM

j6ll committed:
Overhauls the landing page - from simple connection box to a more informative page (#107526)

chriskmnds commented on client/my-sites/promote-post-i2/components/promoted-post-filter/index.tsx:
```diff
@@ -26,6 +50,8 @@ export default function PromotePostTabBar( { tabs, selectedTab }: Props ) {
 		} );
 	};
 	const selectedLabel = tabs.find( ( tab ) => tab.id === selectedTab )?.name;
+	const formattedBalance = '$' + formatNumber( parseFloat( creditBalance ), { decimals: 2 } );
+	const mobileFormattedBalance = '$' + formatNumber( parseFloat( creditBalance ), { decimals: 0 } );
```
We should probably be parsing these with `formatCurrency` instead, from the same package. 

Currencies are more complex; sometimes the value precedes the symbol  (e.g. in EL/DE, it should be `100 $`), or the space is omitted, or the symbol becomes `US$` instead. `formatCurrency` should handle that.

j6ll replied:
Will be taken care of in https://github.com/Automattic/wp-calypso/pull/108083

chriskmnds commented on apps/blaze-dashboard/src/components/generic-header/blaze-plugin-header/index.jsx:
```diff
@@ -10,7 +10,9 @@ const WooBlazeHeader = ( { className = '', children } ) => {
 	return (
 		<header className={ clsx( 'blaze-plugin-header', className ) }>
-			<h2>{ isBlazePlugin ? translate( 'Blaze Ads' ) : translate( 'Advertising' ) }</h2>
+			<h1 className="formatted-header__title wp-brand-font">
```
Hi!`formatted-header__title` feels a little general for referencing this way in code. I think using `FormattedHeader` directly here is safer if that's the style to adapt 🤔

chriskmnds commented on client/my-sites/promote-post-i2/components/campaign-item-details/CampaignDownloadStats.tsx:
```diff
@@ -159,8 +159,7 @@ export default function CampaignDownloadStats( props: Props ) {
 						<Button
-							variant="secondary"
-							className="campaign-item-details__stats-button"
+							className="button posts-list-banner__border-button"
```
I think it's a little suspect that we add a `button` class here to adapt those styles over the components Button 🤔 

(Nothing to do, I can see it used elsewhere from before)

j6ll replied:
Yeah I think in an ideal world, this whole dashboard needs a bit of a code cleanup. It's had a few redesigns and rebuilds stacked on top of each other. So there are some delicate styles and classes in here at the moment. So I tried to tread carefully a bit with this PR and keep some of things in place. As soon as I would change one thing, something else would break elsewhere. So it was a bit of a balancing act :)

chriskmnds replied:
Yeah hearya. The consensus currently regarding dashboard work is to stick with native WP components and styles, unless it is really necessary to deviate. You can read a bit about it here: https://github.com/Automattic/wp-calypso/blob/trunk/client/dashboard/docs/ui-components.md (that's the new dashboard under `my.[wp].com/sites`, etc)

I think the same principle applies to anything concerning admin UI i.e. not just the new hosting dashboard. Less/minimal (literally No) CSS customisations for the most part. :)

j6ll committed:
Revert "show credits expiration date in advertising section (#107901)"

This reverts commit 85c03ac6be6e897187aa1b4fe959e0207ee17129.

chriskmnds commented on apps/blaze-dashboard/src/pages/setup/style.scss:
```diff
@@ -123,3 +123,319 @@
+			margin: 0 0 24px;
+			text-wrap: balance;
+
+			@media (max-width: 768px) {
```
It would be great if we could either use [existing WP mixins/breakpoints](https://wpcalypso.wordpress.com/devdocs/docs/coding-guidelines/css.md#media-queries) or define these a little more centrally somewhere. Right now, these values (`768`, `640`, etc.) are repeated in a few places here.

chriskmnds reviewed (commented):
Heya. Looks great. I left some suggestions below from reviewing the code. I have not tested anything, though. I think the currency formatting is the main critical point. :-)

chriskmnds commented on client/my-sites/promote-post-i2/components/campaigns-total-stats/index.tsx:
```diff
@@ -16,18 +16,23 @@ export default function CampaignsTotalStats( {
+							{ translate( 'Total people reached' ) }
+						</div>
+						<div className="campaigns-total-stats__result">
+							{ formatLargeNumber( totalImpressions || 0 ) }
```
@j6ll We should probably switch these  `formatLargeNumber` cases to `formatNumberCompact` as well. I am not sure if `formatLargeNumber` properly localises the strings. e.g. In EL, 4893000 should render as `4,893 εκ.`

j6ll replied:
Nice! I will switch those references over and remove this custom function. 

Here is the before and after. 

<img width="584" height="108" alt="Screenshot 2026-01-16 at 12 03 29" src="https://github.com/user-attachments/assets/ddd234ea-770b-4efd-a221-836925611c19" />
<img width="568" height="126" alt="Screenshot 2026-01-16 at 12 03 34" src="https://github.com/user-attachments/assets/fd5222a8-9fd6-4b3e-8a11-211236489207" />

j6ll replied:
https://github.com/Automattic/wp-calypso/pull/108169
